.github/workflows/release-mongosh.yml

```diff
@@ -0,0 +1,59 @@
+name: Release mongosh 
+on:
+  workflow_run:
+    workflows: ["CheckQL", "Run Smoke Tests", "evergreen"]
```

Comment:
I am not sure if evergreen is considered a workflow since it seems to be created by a bot/integration(?). Could leave and see what happens or just make it manual-only